materialize-snowflake/snowflake.go

```diff
@@ -452,14 +429,37 @@ func (d *transactor) Store(it *m.StoreIterator) (m.StartCommitFunc, error) {
 					Table:     b.target.Identifier,
 					Query:     mergeIntoQuery,
 					StagedDir: dir,
+					Version:   d.version,
 				}
 			}
 		} else if b.pipeName != "" {
+			var pipeNameParts = strings.Split(b.pipeName, ".")
+
+			// Check to see if a pipe for this version already exists
+			var query = fmt.Sprintf("SHOW PIPES LIKE '%s';", pipeNameParts[len(pipeNameParts)-1])
```

Comment:
We could probably use some caching here to avoid sending this query on every transaction for every binding. Once per version should be sufficient

Reply:
Could we re-use a similar mechanism as the prior `d.pipesInit? Basically just setting it to `true` once we have done the check once, and then not running this branch if it is `true`.

Reply:
@williamhbaker I'm actually going to back down on this. I would prefer if we always check the existence of the pipes. `SHOW PIPES` does not wake up the warehouse, so I think it doesn't hurt in that sense.

The reason is that if for any reason a pipe has been deleted (let's say the user accidentally dropped it), then we would go through transactions, ignoring the fact that the pipe does not exist.